src/SourceBuild/content/eng/pipelines/ci.yml

```diff
@@ -11,16 +13,30 @@ schedules:
 trigger:
   batch: true
   branches:
-   include:
-   - release/*
-   - internal/release/*
+    include:
+    - main
+    - release/*
+    - internal/release/*
+
+pr:
+  branches:
+    include:
+    - main
+    - release/*
 
-pr: none
+variables:
+# enable source-only build for pipelines with the -source-build suffix
+- name: isSourceOnlyBuild
+  value: ${{ in(variables['Build.DefinitionName'], '-source-build') }}
 
 stages:
-- template: templates/stages/vmr-scan.yml
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), eq(variables['Build.Reason'], 'Schedule')) }}:
+  - template: templates/stages/vmr-scan.yml
+    parameters:
+      isSourceOnlyBuild: ${{ variables.isSourceOnlyBuild }}
 
 - template: /src/installer/eng/pipelines/templates/stages/vmr-build.yml
   parameters:
     isBuiltFromVmr: true
-    isLiteBuild: false
+    isLiteBuild: ${{ ne(variables['Build.Reason'], 'Schedule') }}

```

Comment:
Yeah, I can't think of a way to resolve this without two separate pipeline YAML files. Because the triggers need to be different between Lite and Full and triggers can't be conditioned in YAML.

Reply:
Triggers can be conditioned directly in the AzDO UI. We have been doing that in couple places, i.e. runtime. Here's an example:

<img width="960" alt="image" src="https://github.com/dotnet/installer/assets/7412651/5aa97582-1110-47b8-8c12-6c3f4ffe00aa">

Reply:
I agree with Michael and Matt. Part of the reasoning behind having two separate pipelines was to enhance the UX experience. We believed that distinguishing between 'lite' and 'full' builds would prevent confusion. If both were in the same pipeline, a user might mistakenly assume a build was entirely successful, when in fact, they were viewing a 'lite' build and a 'full' build of the same branch had failing components.

Another UX consideration was the cost-effectiveness of manual runs of changes. It's not always necessary to initiate a full build when manually running the pipeline with some modifications. Therefore, having a 'lite' pipeline where users can manually trigger less extensive and cost-effective builds was beneficial.